scripts/watchBackedUpNode.js

```diff
@@ -0,0 +1,132 @@
+var primaryHasToSign = true;
+var validatorSetContract = new web3_2.eth.Contract(
+  require(`${ARTIFACTS_PATH}ValidatorSetAuRa.json`).abi,
+  '0x1000000000000000000000000000000000000001'
+);
+
+async function scanBlocks(depth) {
+    assert(typeof depth === "number");
+    var lastBlock = await web3_2.eth.getBlock("latest");
+    var lastBlockNum = lastBlock.number;
+    assert(typeof lastBlockNum === "number");
+    if (lastBlockNum < depth) {
+        return true;
+    }
+    var startBlockNum = lastBlockNum - depth;
+
+    console.log(`Scanning blocks from ${startBlockNum} to ${lastBlockNum}`);
+
+    for (var i = startBlockNum;  i <= lastBlockNum; i++) {
+        let block = await web3_2.eth.getBlock(i);
+        if (block.author === SIGNER_ADDRESS) {
+            return true;
+        }
+    }
+    return false;
+}
+
+// Starts signing at the secondary node by setting the secondary signer address.
+async function startSecondarySigning() {
+    console.log(`Reserve node at port ${PORT2} starts signing`);
+
+    let password = await readFile(`__dirname${PASSWORD_PATH}`, "UTF-8");
+    assert(typeof password === "string");
+    await provider.send(
+        "parity_setEngineSigner",
+        [ SIGNER_ADDRESS2, password.trim() ]
+    );
+}
+
+// Stops signing at the secondary node by setting the dummy signer address.
+async function stopSecondarySigning() {
+    console.log(`Reserve node at port ${PORT2} stops signing`);
+
+    let password = await readFile(`__dirname${PASSWORD_PATH}`, "UTF-8");
+    assert(typeof password === "string");
+    await provider.send(
+        "parity_setEngineSigner",
+        [ DUMMY_SIGNER_ADDRESS, password.trim() ]
+    );
+}
+
+async function startScan() {
+    var secondaryListening = false;
+    try {
+        secondaryListening = await web3_2.eth.net.isListening();
+    } catch(e) {
+        console.log("Disconnected from secondary");
+    }
+    assert(typeof secondaryListening === "boolean");
+    if (secondaryListening) {
+        let validators = await validatorSetContract.methods.getValidators().call();
+        // Perform failover checks only if the primary is currently a validator.
+        if (validators.indexOf(SIGNER_ADDRESS1) != -1) {
+            var primaryListening = false;
+            try {
+                primaryListening = await web3_1.eth.net.isListening();
+            } catch(e) {
+                console.log("Disconnected from primary");
+            }
+            assert(typeof primaryListening === "boolean");
+            if (!primaryListening) {
+                // Ensure that we (the secondary mode) are still connected to the
+                // network by checking that other validators continued to sign
+                // blocks.
+                let signed = await scanBlocks(validators.length);
+                if (!signed) {
+                    console.log(`Failed to find a block authored by ${SIGNER_ADDRESS}`);
```

Comment:
Should be `SIGNER_ADDRESS1` here

Reply:
Same as above: now there is only one signer address constant.